core/cmd/remote_client.go

```diff
@@ -426,6 +427,27 @@ func (cli *Client) ConfigDump(c *clipkg.Context) (err error) {
 	return err
 }
 
+func (cli *Client) ConfigFileValidate(c *clipkg.Context) (err error) {
+	fileName := c.String("file")
+	data, err := os.ReadFile(fileName)
+	if err != nil {
+		fmt.Printf("Validation failed (%s)\n", err.Error())
+		return err
+	}
+
+	var parsedConfig configV2.Config
+	decoder := toml.NewDecoder(strings.NewReader(string(data)))
+	err = decoder.Decode(&parsedConfig)
+	if err != nil {
+		fmt.Printf("Validation failed (%s)\n", err.Error())
+		return err
+	}
+	// TODO: call ValidateConfig() when PR 6925 gets merged
```

Comment:
You could create a stub for that function (if you know location and the contract) and call it from here. In this case you won't need to modify this code again after 6925 gets merged.

Reply:
If you branch from #6925 and build on top, then you could open this draft against that branch until it gets merged (or merge them together first :shrug:). This would allow you to reuse the flag and app.Before logic as well.

(I'm working on simplifying the `ValidateConfig()` stuff a bit too)